### PR TITLE
feat(MCPE-3848): Send slack notification when Pipeline fails

### PR DESCRIPTION
**Pull request by VisVladimir**

### CODE REVIEW

File: .github/workflows/sng-ingestor-shared-workflow.yml [hunk 1]
```diff
@@ -105,6 +105,16 @@ jobs:
...
 
+      - name: Notify Slack if docker-build fails
+        uses: extenda/actions/slack-notify@v0
+        if: failure()
+        with:
+          text: |
+            *Build failed for ${{ github.repository }}: ${{ github.workflow }}* :heavy_exclamation_mark:
+            Build failed on ${{ github.event_name }} event. Workflow: ${{ github.workflow }}. Job: ${{github.job}}. Run id: <https://github.com/${{ github.repository }}/actions/runs/${{ github.run_id }}|${{ github.run_id }}>
+          channel: selfscan-pipeline-notifications
+          service-account-key: ${{ secrets.secret-auth }}
```
Comment (ivolcov) [inline .github/workflows/sng-ingestor-shared-workflow.yml:116]: Please also add this step to the previous job `validate` and adjust the text accordingly, reflecting the failed job.
Something like:
`Build failed for` should be `Validate failed for` when it is part of `validate` job or
`Staging failed for` when it is part of `staging` job, and so on.

Comment (VisVladimir) [reply .github/workflows/sng-ingestor-shared-workflow.yml]: Done

File: .github/workflows/sng-ingestor-shared-workflow.yml [hunk 2]
```diff
@@ -89,6 +89,16 @@ jobs:
...
           service-account-key: ${{ secrets.gcloud-auth-staging }}
 
+      - name: Notify Slack if staging fails
+        uses: extenda/actions/slack-notify@v0
+        if: failure()
+        with:
+          text: |
+            *Validate failed for ${{ github.repository }}: ${{ github.workflow }}* :heavy_exclamation_mark:
+            Validate failed on ${{ github.event_name }} event. Workflow: ${{ github.workflow }}. Job: ${{ github.job }}. Run id: <https://github.com/${{ github.repository }}/actions/runs/${{ github.run_id }}|${{ github.run_id }}>
+          channel: selfscan-pipeline-notifications
```
Comment (khamooshi) [inline .github/workflows/sng-ingestor-shared-workflow.yml:99]: Please make `channel` to be a input parameter with this channel as the default value.

Comment (VisVladimir) [reply .github/workflows/sng-ingestor-shared-workflow.yml]: Done

File: .github/workflows/sng-ingestor-shared-workflow.yml [hunk 3]
```diff
@@ -89,6 +95,16 @@ jobs:
         with:
           service-account-key: ${{ secrets.gcloud-auth-staging }}
 
+      - name: Notify Slack if staging fails
+        uses: extenda/actions/slack-notify@v0
+        if: failure()
```
Comment (ivolcov) [inline .github/workflows/sng-ingestor-shared-workflow.yml:100]: It should be skipped when the branch is not "master"

File: .github/workflows/sng-proxy-shared-workflow.yml [hunk 4]
```diff
@@ -81,6 +86,16 @@ jobs:
         with:
           service-account-key: ${{ secrets.gcloud-auth-staging }}
 
+      - name: Notify Slack if validate fails
+        uses: extenda/actions/slack-notify@v0
+        if: failure()
```
Comment (ivolcov) [inline .github/workflows/sng-proxy-shared-workflow.yml:91]: Same thing here

Comment (VisVladimir) [reply .github/workflows/sng-proxy-shared-workflow.yml]: Done

File: .github/workflows/sng-ingestor-shared-workflow.yml (hunk 3 above)
Comment (VisVladimir) [reply .github/workflows/sng-ingestor-shared-workflow.yml]: Done

**Review by khamooshi — APPROVED** [review]

**Pull request merged**